app/src/main/java/com/zulip/android/filters/NarrowFilterByDate.java

```diff
@@ -95,4 +97,20 @@ public String getJsonFilter() throws JSONException {
     public String toString() {
         return "{}";
     }
+
+    /**
+     * Checks two dates are of same day or not
+     *
+     * @param date1 Date date1 to be compared with date2
+     * @param date2 Date date2 to be compared with date1
+     * @return boolean
+     */
+    private static boolean isSameDay(Date date1, Date date2) {
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(date1);
+        Calendar calendar2 = Calendar.getInstance();
```

Comment:
This isSameDay will be called very frequently via https://github.com/zulip/zulip-android/blob/a4374c12240d3714710f2dac4376dbecc692a49c/app/src/main/java/com/zulip/android/activities/MessageListFragment.java#L335 
So i'm kinda confused is this getInstance returning a new object? 
If so it would be better to cache these two objects!